external/ag-website-shared/src/components/trial-licence-form/TrialLicenceForm.tsx

```diff
@@ -0,0 +1,274 @@
+
+    const {
+        value: lastName,
+        valueError: validatedLastNameError,
+        handleValueChange: handleLastNameChange,
+    } = useRequiredValidation();
+    const lastNameError = wasValidated && validatedLastNameError ? validatedLastNameError : '';
+
+    const handleFormSubmit: FormEventHandler<HTMLFormElement> = useCallback(
+        async (e) => {
+            e.preventDefault();
+            setWasValidated(true);
+
+            if (validatedEmailError || validatedFirstNameError || validatedLastNameError) {
+                setFormState('error');
+                return;
+            }
+
+            setFormError('');
+            setFormState('loading');
+
+            try {
+                const response = await submitTrialLicenceFormData({ firstName, lastName, email });
+
+                if (response.error) {
+                    setFormState('error');
+                    setFormError(getFormErrorMessage(response.error.message));
+                } else {
+                    setFormState('success');
+                }
+            } catch (e) {
+                console.error(e);
+                setFormError(MESSAGES.formErrorDefault);
+                setFormState('error');
+            }
+        },
+        [validatedEmailError, validatedFirstNameError, validatedLastNameError, firstName, lastName, email]
+    );
+
+    return {
+        formState,
+        formError,
+        emailError,
+        email,
+        handleEmailChange,
+        firstName,
+        firstNameError,
+        handleFirstNameChange,
+        lastName,
+        lastNameError,
+        handleLastNameChange,
+        handleFormSubmit,
+    };
+}
+
+export const TrialLicenceForm: FunctionComponent = () => {
+    const {
+        formState,
+        formError,
+        emailError,
+        email,
+        handleEmailChange,
+        firstName,
+        firstNameError,
+        handleFirstNameChange,
+        lastName,
+        lastNameError,
+        handleLastNameChange,
+        handleFormSubmit,
+    } = useTrialForm();
+    const hasFormError = Boolean(emailError || firstNameError || lastNameError);
+
+    return (
+        <form noValidate className={styles.trailForm} onSubmit={handleFormSubmit}>
+            <div className={styles.inputs}>
+                <div className={classnames('input-field', { 'input-error': firstNameError })}>
+                    <label htmlFor="first-name">First Name</label>
+                    <input
+                        placeholder="First Name"
+                        type="text"
+                        id="first-name"
+                        name="first-name"
+                        value={firstName}
+                        onChange={handleFirstNameChange}
+                        required
+                    />
+
+                    <p className={classnames(!firstNameError && styles.isHidden, 'error')}>First name required</p>
```

Comment:
You can do this with

```suggestion
                    <p className={classnames({ [styles.isHidden]: !firstNameError }, 'error')}>First name required</p>
```